core/transcoder.go

```diff
@@ -41,6 +42,36 @@ func NewUnrecoverableError(err error) UnrecoverableError {
 
 var WorkDir string
 
+func setEffectiveDetectorConfig(md *SegTranscodingMetadata) {
+	aiEnabled := DetectorProfile != nil
+	actualProfile := DetectorProfile
+	if aiEnabled {
+		presetSampleRate := DetectorProfile.(*ffmpeg.SceneClassificationProfile).SampleRate
+		if md.DetectorEnabled && len(md.DetectorProfiles) == 1 {
```

Comment:
Is there a case where there can be more than one DetectorProfiles?

Reply:
Yes, these structures were designed with an aim to have multiple AI capabilities some day.

Reply:
Yep for sure, I was more checking that there was no chance we'd unintentionally hit `md.DetectorEnabled && len(md.DetectorProfiles) > 1` at some point